### ISSUE TITLE
Meteor 1.4: Doesn't load fonts

### ISSUE THREAD

**Issue by dnish** [Dogfalo/materialize#3413]
Hi,
having some trouble with the Materialize package on Meteor 1.4. The Roboto font files aren't found, I get a 404 error.

Example:

```
http://localhost:3000/packages/materialize_materialize/fonts/roboto/Roboto-Medium.woff2
```

**Comment by diegonc** [comment]
Also happens with Meteor 1.3.5.1.

http://localhost:3000/packages/materialize_materialize/fonts/roboto/Roboto-Medium.woff2 -> 404

But adding dist to the URL downloads the font:

http://localhost:3000/packages/materialize_materialize/**dist**/fonts/roboto/Roboto-Medium.woff2 -> Ok

I think this is the same issue as #3411

**Comment by diegonc** [comment]
Oh, it seems the CSS changed and the dist path prefix was removed:

v0.97.6:
```
@font-face {
  font-family: "Roboto";
  src: local(Roboto Regular), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.eot");
  src: url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.eot?") format("embedded-opentype"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.woff2") format("woff2"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.woff") format("woff"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.ttf") format("truetype");
  font-weight: 400;
}
```
v0.97.7:
```
@font-face {
  font-family: "Roboto";
  src: local(Roboto Regular), url("packages/materialize_materialize/fonts/roboto/Roboto-Regular.eot");
  src: url("packages/materialize_materialize/fonts/roboto/Roboto-Regular.eot?") format("embedded-opentype"), url("packages/materialize_materialize/fonts/roboto/Roboto-Regular.woff2") format("woff2"), url("packages/materialize_materialize/fonts/roboto/Roboto-Regular.woff") format("woff"), url("packages/materialize_materialize/fonts/roboto/Roboto-Regular.ttf") format("truetype");
  font-weight: 400;
}
```

**Comment by leanwebtools** [comment]
Yes same issue as #3411 -- how / where would I add 'dist' to the URL to resolve the issue?

**Comment by diegonc** [comment]
I'm not sure.

The change that appears to be the most relevant is "Meteor package now supports Sass".
Before this change, in v0.97.6, the CSS was added in package.js from `dist/css/materialize.css` and the CSS used the font as `../fonts/{...}`; thus pointing into the dist folder.

However in v0.97.7, SASS is used and the files are imported from `sass/{...}` where font-path is defined as
`$roboto-font-path: "../fonts/roboto/" !default;` so apparently `..` points somewhere else in this case.

**Comment by diegonc** [comment]
For instance, what @poetic did in his fork was to change the variables to
```
$roboto-font-path: "/packages/poetic_materialize-scss/dist/font/roboto/" !default;
$icons-font-path: "/packages/poetic_materialize-scss/dist/font/material-design-icons/" !default;
```
but that's not a valid approach here because it would only work in Meteor then.

Maybe he knows better what to do.

**Comment by jhuenges** [comment]
Bump

**Comment by diegonc** [comment]
Ok, in the meantime I've done the following to workaround this issue (because I need the autocomplete feature :smile:):
```
// Temporary workaround for materialize fonts
WebApp.connectHandlers.use("/packages/materialize_materialize/fonts", function (req, res) {
	const url = req.originalUrl.replace("/fonts/", "/dist/fonts/");
	res.statusCode = 301;
	res.setHeader("Location", url);
	res.end();
});
```

**Comment by leanwebtools** [comment]
Thanks, I've dropped this in as a workaround as well. Will keep an eye out for a longer term fix.

**Comment by Metrophobe** [comment]
Just encountered the problem now and am trying to create a package which uses pure css and js (min version) to resolve the issue but so far not there yet im not managing to trigger the js as I am fairly new to packages.

**Comment by jhuenges** [comment]
@Metrophobe to use the code that @diegonc posted, you have to `meteor add webapp` and then copy & paste the code into a server file.

**Comment by polkhovsky** [comment]
So I installed webapp, created a webapp.js file in my /server folder and pasted diegonc's code in it. It doesn't pick up the requests for fonts (I added a console.log to see). A quick pointer would be appreciated.

**Comment by diegonc** [comment]
@polkhovsky, I cannot add anything else but "works for me".

Check this [repo](https://github.com/diegonc/materialize-meteor-font-workaround) for a working brand new project with the workaround.

**Comment by polkhovsky** [comment]
Thank you. It's one of those situations where it by all means should work but doesn't. I appreciate you taking time to create a repo for my reference.

**Comment by Metrophobe** [comment]
@polkhovsky  The fact that the fonts are bundled but are not loading obviously imply that there is something wrong. My simplest workaround and never fails is to 

1. Download the package from the original site (whatever framework it is) 
2. Create a fonts folder inside your public folder inside meteor 
3. Locate the import statements inside the css of the framework, in the event you are using google fonts you are provided this. 
3. Then use @import statements inside the main.css of your meteor app (located in the root). Please remember not to put in the word "public" in your url as it is not required and nothing would load.

Your fonts will now load fine.  At least for the time being while the issue is being fixed, you do not have to stay playing about with packages that should be working but they don't.  I tested it and confirm that this method works fine.

Cheers

**Comment by ArthurRibeiroo** [comment]
Put this code into your main.css:

`@font-face {
  font-family: "Roboto";
  src: local(Roboto Thin), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Thin.eot");
  src: url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Thin.eot?") format("embedded-opentype"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Thin.woff2") format("woff2"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Thin.woff") format("woff"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Thin.ttf") format("truetype");
  font-weight: 200;
}

@font-face {
  font-family: "Roboto";
  src: local(Roboto Light), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Light.eot");
  src: url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Light.eot?") format("embedded-opentype"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Light.woff2") format("woff2"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Light.woff") format("woff"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Light.ttf") format("truetype");
  font-weight: 300;
}

@font-face {
  font-family: "Roboto";
  src: local(Roboto Regular), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.eot");
  src: url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.eot?") format("embedded-opentype"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.woff2") format("woff2"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.woff") format("woff"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Regular.ttf") format("truetype");
  font-weight: 400;
}

@font-face {
  font-family: "Roboto";
  src: url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Medium.eot");
  src: url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Medium.eot?") format("embedded-opentype"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Medium.woff2") format("woff2"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Medium.woff") format("woff"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Medium.ttf") format("truetype");
  font-weight: 500;
}

@font-face {
  font-family: "Roboto";
  src: url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Bold.eot");
  src: url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Bold.eot?") format("embedded-opentype"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Bold.woff2") format("woff2"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Bold.woff") format("woff"), url("packages/materialize_materialize/dist/fonts/roboto/Roboto-Bold.ttf") format("truetype");
  font-weight: 700;
}`

**Comment by dnish** [comment]
I think the best workaround is this for the moment:

Go into your Meteor project folder and create the following folder:

    mkdir -p packages/materialize_materialize/fonts/roboto

Then copy all fonts into it. Everything will work again without modifying any server code.

**Comment by Kostanos** [comment]
Didn't worked in my case @dnish :(

**Comment by keyscores** [comment]
@Dogfalo I noticed you updated the meteor package: https://github.com/Dogfalo/materialize/issues/3057

Though it doesn't seem to solve the typeface issues. There are some duplicated issues:
https://github.com/Dogfalo/materialize/issues/3413
https://github.com/Dogfalo/materialize/issues/3411
https://github.com/Dogfalo/materialize/issues/3281
https://github.com/Dogfalo/materialize/issues/2660

**Comment by keyscores** [comment]
What worked for me was adding this line to main.css

`@import url('https://fonts.googleapis.com/css?family=Roboto:100,400');
`
*Pay attention to the weights you need, like 100, 400 etc.

**Comment by Dogfalo** [comment]
Fixed by #4000
